salt/minion.py

```diff
@@ -2215,34 +2215,23 @@ def manage_schedule(self, tag, data):
         where = data.get('where', None)
         persist = data.get('persist', None)
 
-        if func == 'delete':
-            self.schedule.delete_job(name, persist)
-        elif func == 'add':
-            self.schedule.add_job(schedule, persist)
-        elif func == 'modify':
-            self.schedule.modify_job(name, schedule, persist)
-        elif func == 'enable':
-            self.schedule.enable_schedule()
-        elif func == 'disable':
-            self.schedule.disable_schedule()
-        elif func == 'enable_job':
-            self.schedule.enable_job(name, persist)
-        elif func == 'run_job':
-            self.schedule.run_job(name)
-        elif func == 'disable_job':
-            self.schedule.disable_job(name, persist)
-        elif func == 'postpone_job':
-            self.schedule.postpone_job(name, data)
-        elif func == 'skip_job':
-            self.schedule.skip_job(name, data)
-        elif func == 'reload':
-            self.schedule.reload(schedule)
-        elif func == 'list':
-            self.schedule.list(where)
-        elif func == 'save_schedule':
-            self.schedule.save_schedule()
-        elif func == 'get_next_fire_time':
-            self.schedule.get_next_fire_time(name)
+        funcs = {'delete': self.schedule.delete_job(name, persist),
+                 'add': self.schedule.add_job(schedule, persist),
+                 'modify': self.schedule.modify_job(name, schedule, persist),
+                 'enable': self.schedule.enable_schedule(),
+                 'disable': self.schedule.disable_schedule(),
+                 'enable_job': self.schedule.enable_job(name, persist),
+                 'disable_job': self.schedule.disable_job(name, persist),
+                 'postpone_job': self.schedule.postpone_job(name, data),
+                 'skip_job': self.schedule.skip_job(name, data),
+                 'reload': self.schedule.reload(schedule),
+                 'list': self.schedule.list(where),
+                 'save_schedule': self.schedule.save_schedule(),
+                 'get_next_fire_time': self.schedule.get_next_fire_time(name)}
+
+        # Call the appropriate schedule function
+        if func in funcs:
+            funcs[func]
```

Comment:
This is wrong. You've already called _all of the them_ while creating the `func` dictionary and then not sure what you're trying to do by accessing merely dictionary element. I think you wanted to do this:

```python
functions = {'delete': (name, persist)), 
            'add': (schedule, persist), 
            'modify': (name, schedule, persist)}  # ...and so on
try:
    getattr(self.schedule, functions.get(func))(*params)
except TypeError:
    log.error('Function %s not found', func)
    
```